@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -870,7 +937,25 @@ export class VisibleTileSet {
             }
         });
 
-        this.m_frustumIntersection.updateFrustum();
+        // If elevation is to be taken into account create extended frustum:
+        // (near ~0, far: maxVisibilityRange) that allows to consider tiles that
+        // are far below ground plane and high enough to intersect the frustum.
+        if (elevationRangeSource !== undefined) {
+            const fp = MapViewUtils.getCameraFrustumPlanes(this.m_frustumIntersection.camera);
+            fp.near = this.m_viewRange.minimum;
+            fp.far = this.m_viewRange.maximum;
+            this.m_projectionMatrixOverride.makePerspective(
+                fp.left,
+                fp.right,
+                fp.bottom,
+                fp.top,
+                fp.near,
+                fp.far
+            );
+            this.m_frustumIntersection.updateFrustum(this.m_projectionMatrixOverride);
```

Comment:
Would it be more efficient to compute the min / max height using the ElevationRangeSource. i.e. find the common ancestor of all TileKeys in the scene and get the min / max elevation from that single tile?

Reply:
This approach is presented in the second PR, but it seems that requires more computations and propagating results between calls, so I guess this one is better:
https://github.com/heremaps/harp.gl/pull/762